packages/@mantine/core/src/core/styles-api/use-styles/use-styles.ts

```diff
@@ -46,36 +59,58 @@ export function useStyles<Payload extends FactoryPayload>({
 }: UseStylesInput<Payload>): GetStylesApi<Payload> {
   const theme = useMantineTheme();
   const classNamesPrefix = useMantineClassNamesPrefix();
-  const themeName = (Array.isArray(name) ? name : [name]).filter((n) => n) as string[];
+  const themeName = useMemo(
+    () => (Array.isArray(name) ? name : [name]).filter((n) => n) as string[],
+    [String(name)]
```

Comment:
Putting **String(name)** in the dependency array may not work as intended.
How about putting the value of **name** directly in the dependency array?

Reply:
Sorry, reopened in #6422